### PR TITLE
Hide simpleFind with visibility to remove from tab order.  Fixes: #67563

### LINKED ISSUE #67563: The webview find widget is focusable while hidden
Microsoft/vscode#67509

Repro:

1. Open PR ext description page
2. Focus something, shift+tab until you hit the start of the screen

There are 4 focusable items, close, next match, previous match and the find input.

### PR DESCRIPTION
**Pull request by cleidigh**
Fixes: #67563

@Tyriar 
@mjbvz 

simpleFind widget uses a vertical transition to show and hide itself.  It was visually hidden with active
tabIndexes since display: flex is used and the layout is lost if display: none is used.  The layout
does not like to be changed with the simultaneous transition apparently.  Hence simple solution display: none
toggled with .visible doesn't cut it.

Solution:

- Separate visibility from transition classes
- Use visibility: visible/hidden - maintains flex layout and does not interfere with transition
- For  hide() toggle visible.transition class , delay 200ms then toggle visibility

See issue #67563 four verification (easiest to use NVDA or voiceover readers with screen output on
to track focus even for the hidden states)

### CODE REVIEW

File: src/vs/editor/contrib/find/simpleFindWidget.ts [hunk 1]
```diff
@@ -220,16 +221,20 @@ export abstract class SimpleFindWidget extends Widget {
...
 
 	public hide(): void {
 		if (this._isVisible) {
 			this._isVisible = false;
-
-			dom.removeClass(this._innerDomNode, 'visible');
+			dom.removeClass(this._innerDomNode, 'visible-transition');
 			this._innerDomNode.setAttribute('aria-hidden', 'true');
+			// Need to delay toggling visibility until after Transition, then visibility hidden - removes from tabIndex list
+			setTimeout(() => {
```
Comment (mjbvz) [inline src/vs/editor/contrib/find/simpleFindWidget.ts]: Would it matter if the widget is toggled or shown again before the timeout has fired?  Or will this code still do the correct thing?

**Review by mjbvz — APPROVED** [review]
Thanks for taking a look. The fix looks reasonable to me. Just had a question about the timeout

File: src/vs/editor/contrib/find/simpleFindWidget.ts (hunk 1 above)
Comment (cleidigh) [reply src/vs/editor/contrib/find/simpleFindWidget.ts]: @mjbvz 
Good point.  I based my tweak on the similar setTimeout in reveal() for settings focused. 200ms is right at the edge of human reaction , I certainly can't do anything using my Dragon voice dictation.  That said, it still could be a window.  I slowed down the transition for experimentation.  I think I closed one, probably the only window by not toggling _isVisible until the end of the transition.  With long transition time I could not make any glitches.

**Comment by mjbvz** [comment]
Thanks!

**Pull request merged**